doc/header.md

```diff
@@ -0,0 +1,82 @@
+  - allows user to exchange data in P2P mode;
+- Syncronization between client and server in realtime (close to realtime) and in lazy mode; so applications can work in online and offline (with locally stored data); having bidirectional data sync and hieratchical versioning like git have;
+- Global data structures unification for applications working with [Metarhia](https://github.com/metarhia/Metarhia) technological stack: [GlobalStorage](https://github.com/metarhia/globalgtorage), [Impress](https://github.com/metarhia/impress), [JSTP](https://github.com/metarhia/jstp) and [Console](https://github.com/metarhia/console) through moderated distributed metadata repositories;
+- Ability to work with non-unified data structures (custom schemas), specific to certain subject domain;
+- GlobalStorage provides DAC (data access layer) abstraction, it substitutes ORM but it does not necessarily make maping on the relational model (though RDBMS are also supported);
+- Data structures have global distributed identification system, so data can be inserted anywhere and will not bring ID conflicts;
+- Data reliability is provided by distributed storage facilities, so each data structure should have master server and multiple backup and cache servers; using those servers GlobalStorage supports addressing, versioning and branching.
+
+## Metamodel Definition Language
+
+Using this domain specific language we will describe subject domain in declarative format. To build GUI, API, business-loguic, data structures dynamically in runtime. For example we can build JavaScript prototype and assign it to positional array to access fields by name, so arrays will work like objects.
+
+Example:
+
+```js
+{
+  code: { type: 'string', primary: true },
+  name: {
+    caption: 'City',
+    type: 'string',
+    size: 32,
+    nullable: false,
+    index: { unique: false },
+    master: { dataset: 'Cities', key: 'name' }
+  },
+  birth: 'Date',
+  city: 'string',
+  addresses: {
+    type: { array: 'Address' }
+  },
+  gender: {
+    type: 'char',
+    lookup: { dictionary: { M: 'Male', F: 'Female' } }
+  },
+  age: function() {
+    var difference = new Date() - this.birth;
+    return Math.floor(difference / 31536000000);
+  }
+}
+```
+
+Data types:
+
+- Built-in JavaScript types: string, number, boolean, Date, etc.
+- Global Storage types: id, uid, tree, ip, etc.
+- RDBMS data types: char, int, real, text, money, time, date...
+- Links to other data structures in GlobalStorage
+- Links to other data structures in Application
+
+## JavaScript Query Language
+
+JSQL is a query language for data structures manipulation. JSQL have syntax for:
+filter, projection, dataset join and set operations. We have a separate
+repository for examples and specification:
+[metarhia/JSQL](https://github.com/metarhia/JSQL). Current Implementation can
+be found in [`lib/transformations.js`](lib/transformations.js).
+
+## Distributed Metamodel Repository
```

Comment:
Can we maybe remove this header and the one below, since there is no text/description on the topics?

Reply:
I'd also prefer to remove them since they don't have any useful information, but what about `README.ru.md` which has those headers? Shouldn't they be removed too?

Reply:
This PR doesn't change `README.ru.md`, it may be done separately afterward if someone insists on removing it.